rust/main/agents/relayer/src/server/igp/add_igp_rule.rs

```diff
@@ -0,0 +1,243 @@
+use axum::{extract::State, Json};
+use serde::{Deserialize, Serialize};
+
+use hyperlane_base::server::utils::{ServerResult, ServerSuccessResponse};
+
+use crate::{
+    msg::gas_payment::GasPaymentEnforcer,
+    server::igp::ServerState,
+    settings::{matching_list::MatchingList, GasPaymentEnforcementPolicy},
+};
+
+#[derive(Clone, Debug, Deserialize)]
+pub struct RequestBody {
+    pub policy: GasPaymentEnforcementPolicy,
+    pub matching_list: MatchingList,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct ResponseBody {}
+
+/// Add a new interchain gas payment policy.
+///
+/// Example request
+/// ```json
+/// {
+///     "policy": {
+///         "Minimum": {
+///             "payment": "0x64"
+///         }
+///     },
+///     "matching_list": [
+///         {
+///             "origindomain": 100
+///         }
+///     ]
+/// }
+/// ```
+pub async fn handler(
+    State(mut state): State<ServerState>,
+    Json(payload): Json<RequestBody>,
+) -> ServerResult<ServerSuccessResponse<ResponseBody>> {
+    let RequestBody {
+        policy,
+        matching_list,
+    } = payload;
+
+    tracing::debug!(?policy, ?matching_list, "Adding new IGP rule");
+
+    for (_, gas_enforcer) in state.gas_enforcers.iter_mut() {
```

Comment:
Since we don't need the key of an entry, we can do this:

```suggestion
    for gas_enforcer in state.gas_enforcers.values_mut() {
```